### PR TITLE
Codename: rolodex

### PR DESCRIPTION
**Pull request by raymondng**
@stkem @andrewconner @gmethvin @eishay 

It's a very rudimentary service -- just wiring up the machinery related to new 42 service creation. Hopefully this won't break anything.

For this merge, it's meant to be enabled for dev mode only. Please let me know if you see any issues.

(Yes it's abook and it's easy to type. You are welcome to refer to this service as rolodex ;>)

### CODE REVIEW

**Comment by 2is10** [comment]
`ABookStore` really?

**Comment by 2is10** [comment]
Maybe we could be a little more specific and call it `BarnesAndNoble`.

**Comment by raymondng** [comment]
LOL yes that's a good point -- I didn't see that :)

**Comment by raymondng** [comment]
Sorry I had unintentionally pushed this to master -- my apologies. Please let me know if there are any issues.

File: kifi-backend/modules/abook/app/com/keepit/abook/ABookController.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      val abook = ABook(userId = userId, origin = abookRawInfo.origin, rawInfoLoc = Some(s3Key))
+      val oldVal = abookRepo.findByUserIdAndOriginOpt(userId, abookRawInfo.origin)
+      val entry = oldVal match {
+        case Some(e) => {
+          log.info(s"[upload] old entry for userId=$userId and origin=${abookRawInfo.origin} already exists: $oldVal")
+          e
+        }
+        case None => {
+          val saved = abookRepo.save(abook)
+          log.info(s"[upload] created new abook entry for $userId and ${abookRawInfo.origin} saved entry=$saved")
```
Comment (gmethvin) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookController.scala:41]: textbook use case for a `KestrelCombinator`.

**Comment by gmethvin** [comment]
I wonder if the `ABook` naming convention will confuse future engineers perusing our codebase. Would it be worth the extra letters to call it `AddressBook` or `AddrBook`?

**Comment by 2is10** [comment]
Agreed. Or “contacts”, which is quite common across the industry. Using one
word when possible will help us avoid ever wondering about case and word
separators in different scenarios like class/method names, hostnames, URL
paths, etc.

**Pull request merged**